src/MerQure.Tools/RetryBusService.cs

```diff
@@ -0,0 +1,104 @@
+﻿using MerQure;
+using MerQure.Messages;
+using MerQure.Tools.Configurations;
+using MerQure.Tools.Buses;
+using MerQure.Tools.Messages;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MerQure.Tools
+{
+    public class RetryBusService : IRetryBusService
+    {
+        private readonly IMessagingService _messagingService;
+
+        public RetryBusService(IMessagingService messagingService)
+        {
+            _messagingService = messagingService;
+        }
+
+        public IBus<T> CreateNewBus<T>(RetryStrategyConfiguration configuration) where T : IDelivered
+        {
+            ApplyNewConfiguration(configuration);
+
+            Publisher<T> producer = new Publisher<T>(_messagingService, configuration);
+            RetryConsumer<T> consumer = new RetryConsumer<T>(_messagingService, configuration, producer);
+
+            return new Bus<T>(producer, consumer);
+        }
+
+        private void ApplyNewConfiguration(RetryStrategyConfiguration configuration)
+        {
+            if (configuration.Channels == null || !configuration.Channels.Any())
+            {
+                throw new ArgumentNullException(nameof(configuration.Channels));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The parameter name 'Channels' is not declared in the argument list. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)